### ISSUE TITLE
Add POST /v3/tfidf_result/by_text — encode arbitrary text → TF-IDF neighbor verses (for ICL back-translation)

### ISSUE THREAD

**Issue by woodwardmw** [sil-ai/aqua-api#808]
## Summary

Add a `POST /v3/tfidf_result/by_text` endpoint (plus a batch `by_texts` variant) that takes **raw text** and returns the top-K most similar verses from a TF-IDF assessment's corpus. Today we can query similar verses only by `vref` (`GET /v3/tfidf_result`, the query verse must already be in the corpus) or by a pre-computed 300-dim `vector` (`POST /v3/tfidf_result/by_vector`). There is no way to ask "given this arbitrary text, what are its nearest corpus verses?" — the encoding (TF-IDF → SVD-300) happens outside the API today. This endpoint moves that encoding server-side.

## Motivation

The agent-critique pipeline is moving to an in-context-learning (ICL) back-translation approach: for a draft verse in language X, retrieve the K nearest verses from the existing X-language scripture corpus (by TF-IDF over the **source/X text**) and feed them — together with their pivot-language (e.g. Swahili) translations — as few-shot examples to the translator. The draft verse text is **arbitrary input text**, not a corpus `vref`, so we need text→neighbors. We validated this approach offline and it matches/beats the heavy pipeline against a human back-translation; productionizing it needs this endpoint.

### `POST /v3/tfidf_result/by_text`
Request body:
```json
{
  "assessment_id": 123,          // the TF-IDF assessment whose corpus to search
  "text": "Aga go masyʉ agaa ...",// raw text to encode + match
  "limit": 100,                   // 1..500, default 10
  "reference_id": 106,            // optional: revision to hydrate as reference_text on results
  "exclude_vref": "PRO 1:5",      // optional: drop this vref from results (see Leakage)
  "exclude_book": false           // optional, default false: drop all results in exclude_vref's book
}
```
Response: identical shape to the existing `by_vector` endpoint —
```json
{
  "results": [
    {"id": int, "assessment_id": int, "vref": str, "similarity": float,
     "revision_text": str|null, "reference_text": str|null}
  ],
  "total_count": int
}
```

### `POST /v3/tfidf_result/by_texts` (batch)
```json
{ "assessment_id": 123, "texts": ["...", "..."], "limit": 100,
  "reference_id": 106, "exclude_vrefs": ["PRO 1:5", "PRO 1:6"], "exclude_book": false }
```
Response: `{ "results": [[TfidfResult, ...], ...] }` — one list per input text, mirroring the existing `by_vectors` endpoint. `exclude_vrefs[i]` applies to `texts[i]` (same-length list, or omit). Enforce the same `len(texts) * limit <= 25000` guard as `by_vectors`.

## Encoding (the only new logic)

Encode `text` into the same 300-dim space the corpus vectors live in, then reuse the existing scorer. The full encode chain already exists in **aqua-assessments** and should be ported/reused:

- Rehydration: `aqua-assessments/assessments/tfidf/tfidf_steps/artifacts.py` — `rehydrate_vectorizer(payload)` (lines ~35-52) and `rehydrate_svd(payload)` (lines ~54-99). These reconstruct sklearn `TfidfVectorizer` (word + char) and `TruncatedSVD` directly from the stored artifacts (no pickles).
- Encode chain: `aqua-assessments/assessments/tfidf/app.py` `_encode_and_query` (lines ~567-649):
  ```python
  Xw = word_vec.transform(texts)
  Xc = char_vec.transform(texts)
  X  = normalize(hstack([Xw, Xc]), norm="l2", axis=1)
  vectors = svd.transform(X)           # -> (n, 300)
  ```
  Then score with the **existing** `_score_against_corpus(...)` helper (`tfidf_artifact_routes.py` ~96-119) — the same code path `by_vector`/`by_vectors` already use. Do NOT reimplement scoring.

The stored artifacts are sufficient to do this server-side (confirmed):
- `TfidfVectorizerArtifact` (`database/models.py` ~1310-1331): `vocabulary` (JSONB), `idf` (JSONB), `params` (JSONB: analyzer, ngram_range, lowercase, max_df, min_df) — two rows per run, `kind in ('word','char')`.
- `TfidfSvd` (`models.py` ~1334-1348): `components_npy` (LargeBinary numpy .npy), `dtype`, `n_features`, `n_components`.
- Pull them with the existing `pull_tfidf_artifacts(...)` used by the artifact GET routes.

Model the route after the existing `POST /v3/tfidf_result/by_vector` (`tfidf_artifact_routes.py` ~1150-1202) — it already loads artifacts (for the corpus side), validates, and calls `_score_against_corpus`. The only addition is: rehydrate the vectorizers+SVD and encode `text` → `vector`, then proceed exactly as `by_vector` does.

### Caching
Rehydrating the SVD components matrix (300 × ~20-40k features) per request is the only real cost (~100-200ms). Add a small module-level cache keyed by `assessment_id` holding the rehydrated `(word_vec, char_vec, svd)` so repeated requests to the same assessment skip the decode. A plain dict (optionally TTL-swept) is fine — match whatever caching pattern aqua-assessments' tfidf service already uses (`_cached_artifacts`).

## Leakage / exclusion semantics

- **Default: exclude only the exact verse.** When `exclude_vref` is provided, drop the result whose `vref` matches it (fetch `limit + 1` and drop, so the caller still gets `limit` results) — exactly how the existing `GET /v3/tfidf_result` excludes its query `vref`.
- `exclude_book` (default `false`): when true, drop **all** results whose book == `exclude_vref`'s book (book = the token before the space, e.g. `PRO` from `PRO 1:5`). This is for the rare "re-translating a whole book, don't let its old translation dominate" case.
- We do **not** want to exclude neighboring verses by default — same-book neighbors are useful examples.

## Dependencies

aqua-api `requirements.txt` does not currently include sklearn/scipy. Add:
- `scikit-learn` (pin to match aqua-assessments — currently ~1.6.1; the artifacts store `sklearn_version`, so matching avoids transform drift)
- `scipy` (needed for `scipy.sparse.hstack`)

## Acceptance criteria

1. `POST /v3/tfidf_result/by_text` returns the same response shape as `by_vector`, with `revision_text` (corpus/source) and, when `reference_id` is given, `reference_text` (pivot) hydrated on each result.
2. Encoding a verse's **own text** against the assessment that contains it returns that verse as the top hit with similarity ≈ 1.0 (sanity check that the server-side encode reproduces the stored vector). With `exclude_vref` set to that verse, it is absent and `limit` other results are returned.
3. Batch `by_texts` returns one result list per input text; respects the `len*limit <= 25000` guard.
4. `exclude_book=true` removes all same-book results; default (`false`) keeps them.
5. Latency: first request to an assessment may pay the rehydrate cost; subsequent requests are fast (cache hit).
6. Tests covering: round-trip encode (criterion 2), exclude_vref, exclude_book, batch, and a missing-artifacts 404.

## References (existing code to reuse / model after)
- `assessment_routes/v3/tfidf_artifact_routes.py`: `by_vector` ~1150-1202, `by_vectors` ~1205-1279, `_score_against_corpus` ~96-119, `pull_tfidf_artifacts`.
- `assessment_routes/v3/results_query_routes.py`: `get_tfidf_result` (vref endpoint, self-exclusion pattern) ~1379-1509.
- `database/models.py`: artifact schema ~1278-1348.
- aqua-assessments `assessments/tfidf/tfidf_steps/artifacts.py` (rehydration) and `assessments/tfidf/app.py` `_encode_and_query` ~567-649 (the encode chain to port).
